src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -6,40 +6,38 @@
 using System.Linq;
 using AutoRest.CSharp.Common.Input;
 using AutoRest.CSharp.Generation.Types;
-using AutoRest.CSharp.Generation.Writers;
 using AutoRest.CSharp.Input;
 using AutoRest.CSharp.Input.Source;
 using AutoRest.CSharp.Mgmt.Decorator;
-using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Builders;
 using Azure.ResourceManager.Models;
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
     internal class ObjectTypeProperty
     {
-        public ObjectTypeProperty(FieldDeclaration field, InputModelProperty inputModelProperty, ObjectType enclosingType)
+        public ObjectTypeProperty(FieldDeclaration field, InputModelProperty? inputModelProperty)
             : this(declaration: new MemberDeclarationOptions(field.Accessibility, field.Name, field.Type),
                   parameterDescription: field.Description?.ToString() ?? string.Empty,
                   isReadOnly: field.Modifiers.HasFlag(FieldModifiers.ReadOnly),
                   schemaProperty: null,
                   isRequired: field.IsRequired,
-                  valueType: field.ValueType,
+                  valueType: null,
```

Comment:
would this cause a breaking here because after the consolidation we will always use the FieldDeclaration to create a property, and there are properties that have different types for `Declaration.Type` and `ValueType`?
If we assign null to valueType, they will always have the same value as declaration type.

Reply:
No, because `JsonPropertySerialization`/`XmlPropertySerialization` objects are created from `InputTypeProperty` directly (see `SerializationBuilder.CreateJsonPropertySerializationFromInputModelProperty`)